### PR TITLE
Tolerate non-default pandas Index in `Scenario.years_active()`

### PR DESCRIPTION
**Pull request by OFR-IIASA**
This PR introduces the updated syntax to resolve the issue as described in Issue #618 

## How to review

Check code adjustments and ensure all tests pass.

## PR checklist

- [x] Continuous integration checks all ✅
- [x] ~Add or expand tests;~ coverage checks both ✅
- ~Add, expand, or update documentation.~ N/A, syntax correction, no change in behavior
- ~Update release notes.~

### CODE REVIEW

**Comment by OFR-IIASA** [comment]
@khaeru we only recently added the test which are currently failing, but it seems that these are now showing an error. Any suggestions on how to adjust the tests so that lint-checks pass?

**Comment by khaeru** [comment]
> @khaeru we only recently added the test which are currently failing, but it seems that these are now showing an error. Any suggestions on how to adjust the tests so that lint-checks pass?

So the tests per se are fine, passing. What's happened is that the mypy checks of typing have begun to fail:
```
message_ix/tests/test_feature_vintage_and_active_years.py:26: error: Incompatible types in assignment (expression has type "ndarray[Any, dtype[Any]]", variable has type "List[ndarray[Any, dtype[Any]]]")  [assignment]
message_ix/tests/test_feature_vintage_and_active_years.py:29: error: Argument 1 to "filter" has incompatible type overloaded function; expected "Callable[[Tuple[Any, Any]], TypeGuard[Union[_SumT, Literal[0]]]]"  [arg-type]
message_ix/tests/test_feature_vintage_and_active_years.py:29: error: No overload variant of "__getitem__" of "list" matches argument type "Tuple[int, slice]"  [call-overload]
message_ix/tests/test_feature_vintage_and_active_years.py:29: note: Possible overload variants:
message_ix/tests/test_feature_vintage_and_active_years.py:29: note:     def __getitem__(self, SupportsIndex) -> ndarray[Any, dtype[Any]]
message_ix/tests/test_feature_vintage_and_active_years.py:29: note:     def __getitem__(self, slice) -> List[ndarray[Any, dtype[Any]]]
Found 3 errors in 1 file (checked 48 source files)
```

Usually this is due to a new version of mypy, or updated type hints for one of the upstream packages we use, e.g. pandas. I can investigate, although not before Friday.

**Comment by khaeru** [comment]
Also, let's use a more descriptive title for PR. "issue #618" is only intelligible if one happens to remember what that issue is about. Few do now, and in the future no one will.

**Comment by khaeru** [comment]
Also removed the review request, since the PR is not ready for review:
- Checklist is not complete.
- CI checks are not passing.

**Comment by khaeru** [comment]
FYI @OFR-IIASA I opened #627 to address the typing error, which @GamzeUnlu95 also encountered in https://github.com/iiasa/message_ix/pull/623#issuecomment-1154924231. I asked for a review but haven't got one yet. If you'd like to approve that, then I can merge; then you can rebase here and it should pass.

File: RELEASE_NOTES.rst [hunk 1]
```diff
@@ -16,6 +16,7 @@ Migration notes
 All changes
 -----------
 
+- Correct functionality-syntax of :meth:`.vintage_and_active_years` (:pull:`623`).
 - Extend functionality of :meth:`.vintage_and_active_years`; add aliases
   :meth:`.yv_ya`, :meth:`.ya`, and :attr:`.y0` (:pull:`572`).
```
Comment (khaeru) [inline RELEASE_NOTES.rst:21]: This is fine.

Just FYI, another option, when we have 2+ PRs in the same release that combine to create/extend the same feature(s), is to mention them all after the same bullet e.g.:
```rest
- Extend functionality of … (:pull:`572`, :pull:`623`)
```

**Review by khaeru — APPROVED** [review]

**Pull request merged**